app/assets/javascripts/discourse/app/components/d-editor.js

```diff
@@ -351,6 +363,26 @@ export default Component.extend(TextareaTextManipulation, {
     }
   },
 
+  @bind
+  onBeforeInputSmartList(event) {
+    // This inputType is much more consistently fired in `beforeinput`
+    // rather than `input`.
+    const isNewLine = event.inputType === "insertLineBreak";
+    if (isNewLine) {
+      this.handleSmartListAutocomplete = true;
+    } else {
+      this.handleSmartListAutocomplete = false;
+    }
```

Comment:
```suggestion
    this.handleSmartListAutocomplete = isNewLine;
```

Reply:
Don't come in here with your fancy last minute refactors CVX :P